anvil/tests/test_models.py

```diff
@@ -59,25 +64,31 @@ def test_model_construction(layer_idx, n_blocks, lattice_length_half, hidden_sha
 
 def layer_independence_test(model_spec):
     """Check that each layer's parameters are updated independently."""
-
-    # Collect over these layers
-    model = API.model_to_load(**model_spec)
-    layer1, layer2 = [layer for layer in model]
-
-    layer2_copy = deepcopy(layer2)
+    model = iter(API.model_to_load(**model_spec))
+    model_copy = deepcopy(model)
 
     # Update parameters in first layer
-    valid_key, valid_tensor = next(iter(layer1.state_dict().items()))
-    update = {valid_key: torch.rand_like(valid_tensor)}
+    layer1 = next(model)
+    update = {}
+    for valid_key, valid_tensor in layer1.state_dict().items():
+        update[valid_key] = torch.rand_like(valid_tensor)
     layer1.load_state_dict(update, strict=False)
 
-    # Check that second layer is unchanged
+    # Check that this is different from the copy
+    layer1_copy = next(model_copy)
+    for original, copy in zip(layer1.parameters(), layer1_copy.parameters()):
+        assert not torch.allclose(original, copy)
+
+    # Now check that the other layers are unchanged
     # NOTE: may be safer to iterate over shared keys
-    for original, copy in zip(layer2.parameters(), layer2_copy.parameters()):
-        assert torch.allclose(original, copy)
+    for layer, layer_copy in zip(model, model_copy):
+        for original, copy in zip(layer.parameters(), layer_copy.parameters()):
+            if not torch.allclose(original, copy):
+                raise LayersNotIndependentError(
+                    "Parameters are being shared amongst layers that should be independent."
```

Comment:
was there a reason you couldn't except an assertion error?

Reply:
no